File: M8/Exemplos/App_eXperience_Kotlin/app/src/main/java/com/example/e1_elgin_kotlin/FragmentPrinterText.kt
package com.example.e1_elgin_kotlin

import android.app.AlertDialog
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.fragment.app.Fragment
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.util.*

class FragmentPrinterText : Fragment() {
    lateinit var buttonPrinter: Button
    lateinit var buttonPrinterXMLNFCe: Button
    lateinit var buttonPrinterXMLSAT: Button

    lateinit var radioGroupAlign: RadioGroup
    lateinit var buttonRadioCenter: RadioButton

    lateinit var editTextInputMessage: EditText

    lateinit var spinnerFontFamily: Spinner
    lateinit var spinnerFontSize: Spinner

    lateinit var checkBoxIsBold: CheckBox
    lateinit var checkBoxIsUnderLine: CheckBox
    lateinit var checkBoxIsCutPaper: CheckBox

    var typeOfAlignment:String = "Centralizado"
    var typeOfFontFamily:String = "FONT A"
    var typeOfFontSize:Int = 17

    var xmlNFCe:String = "xmlnfce"
    var xmlSAT:String = "xmlsat"

    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View {

        val view = inflater.inflate(R.layout.activity_fragment_printer_text, container, false)

        radioGroupAlign = view.findViewById(R.id.radioGroupAlign)
        buttonRadioCenter = view.findViewById(R.id.radioButtonCenter)
        spinnerFontFamily = view.findViewById(R.id.spinnerFontFamily)
        spinnerFontSize = view.findViewById(R.id.spinnerFontSize)
        checkBoxIsBold = view.findViewById(R.id.checkBoxBold)
        checkBoxIsUnderLine = view.findViewById(R.id.checkBoxUnderline)
        checkBoxIsCutPaper = view.findViewById(R.id.checkBoxCutPaper)

        editTextInputMessage = view.findViewById(R.id.editTextInputMessage)
        editTextInputMessage.setText("ELGIN DEVELOPERS COMMUNITY")

        buttonPrinter = view.findViewById(R.id.buttonPrinterText)
        buttonPrinterXMLNFCe = view.findViewById(R.id.buttonPrinterNFCe)
        buttonPrinterXMLSAT = view.findViewById(R.id.buttonPrinterSAT)

        buttonRadioCenter.isChecked = true
        radioGroupAlign.setOnCheckedChangeListener { group, checkedId ->
            if (checkedId == R.id.radioButtonLeft)
                typeOfAlignment = "Esquerda"

            if (checkedId == R.id.radioButtonCenter)
                typeOfAlignment = "Centralizado"

            if (checkedId == R.id.radioButtonRight)
                typeOfAlignment = "Direita"
            println(typeOfAlignment)
        }

        spinnerFontFamily.onItemSelectedListener = object: AdapterView.OnItemSelectedListener {
            override fun onItemSelected(
                    parent: AdapterView<*>?,
                    view: View?,
                    position: Int,
                    id: Long
            ) {
                typeOfFontFamily = parent?.getItemAtPosition(position).toString()
                println(typeOfFontFamily)
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {
                TODO("Not yet implemented")
            }
        }

        spinnerFontSize.onItemSelectedListener = object: AdapterView.OnItemSelectedListener {
            override fun onItemSelected(
                    parent: AdapterView<*>?,
                    view: View?,
                    position: Int,
                    id: Long
            ) {
                var fontSize = parent?.getItemAtPosition(position).toString()
                typeOfFontSize = fontSize.toInt()
                println(typeOfFontSize)
            }
            override fun onNothingSelected(parent: AdapterView<*>?) {
                TODO("Not yet implemented")
            }
        }

        buttonPrinter.setOnClickListener() {
            printText()
        }

        buttonPrinterXMLNFCe.setOnClickListener(){
            printXmlNFCe()
        }

        buttonPrinterXMLSAT.setOnClickListener(){
            printXmlSAT()
        }

        return view
    }

    private fun printText() {
        val mapValues: MutableMap<String, Any> = HashMap()

        if (editTextInputMessage.text.toString() == "") {
            alertMessageStatus("Alert", "Campo Mensagem vazio.")
        } else {
            mapValues["text"] = editTextInputMessage.text.toString()
            mapValues["align"] = typeOfAlignment
            mapValues["font"] = typeOfFontFamily
            mapValues["fontSize"] = typeOfFontSize
            mapValues["isBold"] = checkBoxIsBold.isChecked
            mapValues["isUnderline"] = checkBoxIsUnderLine.isChecked

            PrinterMenu.printer.imprimeTexto(mapValues)

            jumpLine()
            if (checkBoxIsCutPaper.isChecked) cutPaper()
        }
    }

    @Throws(IOException::class)
    private fun printXmlNFCe() {
        val stringXMLNFCe: String

        val ins: InputStream = context?.resources?.openRawResource(
                requireContext().resources.getIdentifier(
                        xmlNFCe,
                        "raw",
                        requireContext().packageName
                )
        ) !!

        val br = BufferedReader(InputStreamReader(ins))
        val sb = StringBuilder()
        var line: String? = null
        try {
            line = br.readLine()
        } catch (e: IOException) {
            e.printStackTrace()
        }
        while (line != null) {
            sb.append(line)
            sb.append(System.lineSeparator())
            line = br.readLine()
        }
        stringXMLNFCe = sb.toString()

        val mapValues: MutableMap<String?, Any?> = HashMap()

        mapValues["xmlNFCe"] = stringXMLNFCe
        mapValues["indexcsc"] = 1
        mapValues["csc"] = "CODIGO-CSC-CONTRIBUINTE-36-CARACTERES"
        mapValues["param"] = 0

        PrinterMenu.printer.imprimeXMLNFCe(mapValues)

        jumpLine()
        if (checkBoxIsCutPaper.isChecked) cutPaper()
    }

    @Throws(IOException::class)
    private fun printXmlSAT() {
        val stringXMLSat: String

        val ins: InputStream = context?.resources?.openRawResource(
                requireContext().resources.getIdentifier(
                        xmlSAT,
                        "raw",
                        requireContext().packageName
                )
        )!!

        val br = BufferedReader(InputStreamReader(ins))
        val sb = java.lang.StringBuilder()
        var line: String? = null
        try {
            line = br.readLine()
        } catch (e: IOException) {
            e.printStackTrace()
        }
        while (line != null) {
            sb.append(line)
            sb.append(System.lineSeparator())
            line = br.readLine()
        }
        stringXMLSat = sb.toString()

        val mapValues: MutableMap<String?, Any?> = HashMap()

        mapValues["xmlSAT"] = stringXMLSat
        mapValues["param"] = 0

        PrinterMenu.printer.imprimeXMLSAT(mapValues)

        jumpLine()
        if (checkBoxIsCutPaper.isChecked) cutPaper()
    }

    private fun jumpLine() {
        val mapValues: MutableMap<String?, Any?> = HashMap()
        mapValues["quant"] = 10
        PrinterMenu.printer.AvancaLinhas(mapValues)
    }

    private fun cutPaper() {
        val mapValues: MutableMap<String?, Any?> = HashMap()
        mapValues["quant"] = 10
        PrinterMenu.printer.cutPaper(mapValues)
    }

    private fun alertMessageStatus(titleAlert: String, messageAlert: String) {
        val alertDialog = AlertDialog.Builder(context)

        alertDialog.setTitle(titleAlert)
        alertDialog.setMessage(messageAlert)
        alertDialog.setPositiveButton("OK"){ dialog, which -> dialog.dismiss()}
        alertDialog.create()
        alertDialog.show()
    }
}

File: Exemplos/App_eXperience_XamarinAndroid/Impressora/FragmentPrinterText.cs
using Android.Support.V4.App;
using Android.OS;
using Android.Views;
using Android.Widget;
using Android.Graphics;
using System;
using System.Collections.Generic;
using Xamarin.Essentials;

namespace M8
{

    public class FragmentPrinterText : Fragment
    {
        EditText editMensagem;
        RadioButton radioEsquerda, radioCentralizado, radioDireita;
        Spinner spinnerFontFamily, spinnerFontSize;
        CheckBox checkboxNegrito, checkboxSublinhado, checkboxCutPaper;
        Button btnImprimirTexto, btnNFCE, btnSAT;

        private string nomeXmlSat = "xmlsat";
        private string nomeXmlNFCE = "xmlnfce";
        private string xmlSat;
        private string xmlNFCE;

        Dictionary<string, string> parametros;

        public override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            // Create your fragment here
           
        }

        public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
        {
            base.OnCreateView(inflater, container, savedInstanceState);
            // Use this to return your custom view for this Fragment
            View fragmentView = inflater.Inflate(Resource.Layout.fragmentprintertext, container, false);

            xmlSat = CarregarArquivo(nomeXmlSat);
            xmlNFCE = CarregarArquivo(nomeXmlNFCE);
           
            //Atribuição das Views - FragmentPrinterText

            //EditTexts
            editMensagem = fragmentView.FindViewById<EditText>(Resource.Id.editMensagem);

            //Radios
            radioEsquerda = fragmentView.FindViewById<RadioButton>(Resource.Id.radioEsquerda);
            radioCentralizado = fragmentView.FindViewById<RadioButton>(Resource.Id.radioCentralizado);
            radioDireita = fragmentView.FindViewById<RadioButton>(Resource.Id.radioDireita);

            //Spinners
            spinnerFontFamily = fragmentView.FindViewById<Spinner>(Resource.Id.spinnerFontFamily);
            spinnerFontSize = fragmentView.FindViewById<Spinner>(Resource.Id.spinnerFontSize);

            //Checkboxes
            checkboxNegrito = fragmentView.FindViewById<CheckBox>(Resource.Id.checkboxNegrito);
            checkboxSublinhado = fragmentView.FindViewById<CheckBox>(Resource.Id.checkboxSublinhado);
            checkboxCutPaper = fragmentView.FindViewById<CheckBox>(Resource.Id.checkboxCutPaper);

            //Buttons
            btnImprimirTexto = fragmentView.FindViewById<Button>(Resource.Id.btnImprimirTexto);
            btnNFCE = fragmentView.FindViewById<Button>(Resource.Id.btnNFCE);
            btnSAT = fragmentView.FindViewById<Button>(Resource.Id.btnSAT);

            InitialState();


            InitViewsFunc();
            
          

            return fragmentView;
        }

        private void InitialState()
        {
            editMensagem.Text = "ELGIN DEVELOPER COMMUNITY";
            radioCentralizado.Checked = true;
            spinnerFontFamily.SetSelection(0);
            spinnerFontSize.SetSelection(0);
        }
        private void InitViewsFunc()
        {
           


            btnImprimirTexto.Click += delegate
            {
                if (editMensagem.Text == "")
                {
                    alert("Campo mensagem vazio!");
                }
                else
                {
                    parametros = new Dictionary<string, string>();

                    parametros.Add("text", editMensagem.Text);

                    if (radioEsquerda.Checked == true) parametros.Add("align", "Esquerda");
                    else if (radioCentralizado.Checked == true) parametros.Add("align", "Centralizado");
                    else parametros.Add("align", "Direita");


                    if (spinnerFontFamily.SelectedItem.ToString() == "Fonte A") parametros.Add("font", "FONT A");
                    else parametros.Add("font", "FONT B");

                    parametros.Add("fontSize", spinnerFontSize.SelectedItem.ToString());

                    if (checkboxNegrito.Checked == true) parametros.Add("isBold", "true");
                    else parametros.Add("isBold", "false");

                    if (checkboxSublinhado.Checked == true) parametros.Add("isUnderline", "true");
                    else parametros.Add("isUnderline", "false");


                    Printer.imprimeTexto(parametros);
                    Printer.AvancaLinhas(10);
                    if (checkboxCutPaper.Checked) Printer.cutPaper(10);

                    
                }

            };

            btnNFCE.Click += delegate
            {
                parametros = new Dictionary<string, string>();

                parametros.Add("xmlNFCe", xmlNFCE);
                parametros.Add("indexcsc", "1");
                parametros.Add("csc", "CODIGO-CSC-CONTRIBUINTE-36-CARACTERES");
                parametros.Add("param", "0");
                Printer.imprimeXMLNFCe(parametros);
                Printer.AvancaLinhas(10);
                if (checkboxCutPaper.Checked) Printer.cutPaper(10);
            };

            btnSAT.Click += delegate
            {
                parametros = new Dictionary<string, string>();

                parametros.Add("xmlSAT",xmlSat);
                parametros.Add("param","0");
                Printer.imprimeXMLSAT(parametros);
                Printer.AvancaLinhas(10);
                if (checkboxCutPaper.Checked) Printer.cutPaper(10);
            };
        }

        private void alert(String message)
        {
            Android.App.AlertDialog alertDialog = new Android.App.AlertDialog.Builder(Impressora.impressoraActivity).Create();
            alertDialog.SetTitle("Alerta");
            alertDialog.SetMessage(message);
            alertDialog.SetButton("OK", delegate
            {
                alertDialog.Dismiss();
            });
            alertDialog.Show();
        }

        protected string CarregarArquivo(string nomeArquivo)
        {
            var stream = this.Context.Resources.OpenRawResource(this.Context.Resources.GetIdentifier(nomeArquivo, "raw", this.Context.PackageName));
            string conteudoArquivo = "";
            using (var reader = new System.IO.StreamReader(stream))
            {
                conteudoArquivo = reader.ReadToEnd();
            }
            return conteudoArquivo;

        }
    }


}